rustls-platform-verifier/src/verification/windows.rs

```diff
@@ -521,7 +565,24 @@ impl Verifier {
             .chain(Some(0))
             .collect();
 
-        let cert_chain = store.new_chain_in(&primary_cert, now)?;
+        let mut cert_chain = store.new_chain_in(&primary_cert, now)?;
+
+        // If we have extra roots and the building the chain gave us an error, we try to build a
+        // new one with the extra roots.
+        if !self.extra_roots.is_empty()
+            // SAFETY: The pointer is guaranteed to be non-null.
+            && unsafe { *cert_chain.inner.as_ptr() }
+                .TrustStatus
+                .dwErrorStatus & CERT_TRUST_IS_PARTIAL_CHAIN != 0
```

Comment:
Nit: Please extract this out into its own binding for clarity. For example:
```rust
let build_status = unsafe { cert_chain.inner.as_ref().TrustStatus.dwErrorStatus };
if build_status & CERT_TRUST_IS_PARTIAL_CHAIN != 0 && !self.extra_roots.is_empty() {
    ...
}
```

Reply:
That's much cleaner, thank you